app/controllers/api/v2/webhooks_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    class WebhooksController < V2::BaseController
+      include Api::Version2
+      include ForemanWebhooks::Controller::Parameters::Webhook
+      include Foreman::Controller::TemplateImport
+
+      before_action :find_resource, only: %i[show update destroy]
+
+      api :GET, '/webhooks/', N_('List Webhooks')
+      param_group :search_and_pagination, ::Api::V2::BaseController
+      def index
+        @webhooks = resource_scope_for_index
+      end
+
+      api :GET, '/webhooks/:id', N_('Show Webhook details')
+      param :id, :identifier, required: true
+      def show; end
+
+      def_param_group :webhook do
+        param :webhook, Hash, action_aware: true, required: true do
+          param :name, String, required: true
+          param :target_url, String, required: true
+          param :events, Array, required: true
+          param :payload_template_id, :identifier
```

Comment:
Do we actually need to have multiple events to be associated with a webhook? I'd rather keep templates simple, I understand that ERB is powerful and probably some things could be generic. To the extreme, one webhook could catch them all - that's probably quite useful use case.

Reply:
For early testing purposes I've added a possibility to set content-type and http method via UI/API.

Now I'm going to add the rest in a new commit. The bad thing about is that it will be a bit messy with the database: you'll need to rollback the latest migration, then checkout and then migrate. For the sake of keeping addition of new columns in one migration. Although I don't mind if it will be one or more migrations.

Reply:
Yeah rollbacking is something we can't avoid at some point.